tests/test_exceptions.py

```diff
@@ -248,6 +248,11 @@ def pycatch(exctype, f, *args):  # noqa: ARG001
     assert str(excinfo.value) == "this is a helper-defined translated exception"
 
 
+# TODO: Investigate this crash
+@pytest.mark.skipif(
```

Comment:
In similar situations before I created a bug, tracking information from

the last GHA log that still worked

the first GHA log that was broken

I can do that a little later, then we can reference that bug here.

Reply:
Logs are deleted after a time; I guess you mean copy that info here then link here?

Reply:
> I guess you mean copy that info here then link here?

Yes

(trying to find an example now; writing from a train with choppy internet connectivity ...)

Reply:
Here is an example: https://github.com/pybind/pybind11/issues/4889

I'm pointing to github.com/rwgk, where I checked in copies of the logs.

I can do the same for this PR. We can simple use a comment here for the equivalent of the 4889 PR description.

Reply:
Okay, will it be ready to merge once you do that?

Reply:
I see this in the log:

```
C++ Info: Clang 18.1.0rc C++17 __pybind11_internals_v5_msvc_mscver1939_debug__ PYBIND11_SIMPLE_GIL_MANAGEMENT=False
```

Would I had in mind is to produce (made up):

```
C++ Info: Clang 18.1.0rc MSVC 193933522 C++17 __pybind11_internals_v5_msvc_mscver1939_debug__ PYBIND11_SIMPLE_GIL_MANAGEMENT=False
```

This could well help in the future easily mining logs programmatically (`grep`).

But if you're strongly opposed for some reason, what you have seems acceptable.

Reply:
I'd make this specific to the Clang version, e.g. "Clang 18".

and for the reason: `"See pybind/pybind11#5062 for background."`

Good to merge with that.

Reply:
I believe Clang defines the MSVC macros for compatibility, just like it defined the GCC macros for compatibility on Unix. The value they are defined to isn't really relevant, you just need to know the clang version.

It's happening on at least 16 and 18, so I don't think it's clang version specific. I think it's just installing clang from chocolaty, not sure what versions are available, but 16 is pre-installed on the runner and our setup action is updating it to 18.

Reply:
> The value they are defined to isn't really relevant

They are informative. Could make the difference between being confused about what path is taken and being completely clear, when debugging.

Reply:
Seeing “MSVC” when MSVC isn’t being used is more likely to be confusing IMO. And why are we printing out the MSVC compat values and not the GCC ones? Information that is misleading and redundant may be worse than no information at all.

The purpose of the compiler info is to tell the user what compiler is being used and to help with test skips. Adding MSVC doesn’t tell the use what compiler is being used and will confuse any tests that actually need to be skipped on MSVC (for example).